src/crawlers/browser_crawler.js

```diff
@@ -194,21 +195,24 @@ class BrowserCrawler extends BasicCrawler {
         request.loadedUrl = await page.url();
     }
 
-    async _createContextFunction(pluginOptions) {
+    async extendLaunchContext(launchContext) {
+        const crawlingContext = this.crawlingContexts.get(launchContext.contextId);
+
+        launchContext.crawlingContext = crawlingContext;
+
         let session;
-        let proxyInfo;
-        let proxyUrl;
 
         if (this.sessionPool) {
             session = await this.sessionPool.getSession();
+            launchContext.session = session;
```

Comment:
any reason not to use `crawlingContext.session`?

Reply:
maybe we don't even need to save the session to `launchContext`, when we have it in `launchContext.crawlingContext.session` anyway?

Reply:
We don't know here if this session is going to be used in the launch context or not. Actually thinking about it one more time we know, so we can probably put it straight to the `crawlingContext`.